### PR TITLE
Add VSphereVM conditions

### PR DESCRIPTION
**Pull request by fabriziopandini**
**What this PR does / why we need it**:
This PR adds conditions to the VSphereVM kind

**Special notes for your reviewer**:
This PR builds on top of https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/pull/934, so please consider only the last commit

**Release note**:
```release-note
The VSphereVM kind now has a `Status.Conditions` field providing evidence about the current state of the objects.
```

/hold
/assign @yastij

### CODE REVIEW

File: pkg/services/govmomi/service.go [hunk 1]
```diff
@@ -118,6 +130,7 @@ func (vms *VMService) ReconcileVM(ctx *context.VMContext) (vm infrav1.VirtualMac
 		return vm, err
 	}
 
+	conditions.MarkTrue(ctx.VSphereVM, infrav1.VMProvisionedCondition)
```
Comment (fabriziopandini) [inline pkg/services/govmomi/service.go:133]: This is probably redundant given we are doing the same in `reconcileNormal`...

Comment (fabriziopandini) [reply pkg/services/govmomi/service.go]: removed

**Comment by fabriziopandini** [comment]
/hold cancel
rebased after #934 merged

File: api/v1alpha3/condition_consts.go [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+// Conditions and condition Reasons for the VSphereVM object
+
+const (
+
+	// VMProvisionedCondition documents the status of the provisioning of a VSphereVM.
+	VMProvisionedCondition clusterv1.ConditionType = "VMProvisioned"
+
+	// CloningReason documents (Severity=Info) a VSphereVM currently executing the clone operation.
+	CloningReason = "Cloning"
```
Comment (yastij) [inline api/v1alpha3/condition_consts.go:29]: one question, do we want condition to be on going or in the past

File: api/v1alpha3/condition_consts.go [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+	// CloningFailedReason (Severity=Warning) documents a VSphereVM controller detecting
+	// an error while provisioning; the reconcile loop will automatically retry the operation,
+	// but a user intervention might be required to fix the problem.
+	CloningFailedReason = "CloningFailed"
+
+	// PoweringOnReason documents (Severity=Info) a VSphereVM currently executing the power on sequence.
+	PoweringOnReason = "PoweringOn"
```
Comment (yastij) [inline api/v1alpha3/condition_consts.go:37]: same

File: api/v1alpha3/condition_consts.go (hunk 2 above)
Comment (fabriziopandini) [reply api/v1alpha3/condition_consts.go:29]: VMProvisining happens only one time at the beginning of the machine lifecycle, so IMO it is fine to use past

**Comment by fabriziopandini** [comment]
@yastij rebased

**Pull request merged**